Proper Copilots in depth guide (#123)
Credal-ai/fern-docs: fern/docs/pages/guides/copilots/in-depth-guide.mdx

# AI Copilots In-Depth Guide

_This is a comprehensive in depth guide for Copilots. If you’re just getting started, check out [Getting Started With AI Copilots](/copilots/quick-start)._

Credal’s AI Copilots empower users to set up dedicated assistants for a wide range of use cases, from customer support to contract review.

Copilots can assist with any task that combines AI and data. They are designed to be experts on the data and context you provide. They combine AI with your data to provide accurate, context-aware responses, while citing their sources.

**Contents:**

## 1. **Setting up a Copilot**

_It’s helpful to have a good sense of what you want your copilot to do from the start, so you can set it up to achieve its objectives._

To get started:

1. On your left sidebar, Select “Copilots”:

![copilots_lhs](/docs/assets/copilots/copilots_lhs.png)

1. Select the “Create new copilot” button.

![create_new_copilot_button](/docs/assets/copilots/create_new_copilot_button.png)

1. Give your copilot a name and add a description of its function in the pop-out form. You can always come back and edit these later.

>    _Descriptions should be detailed enough to describe what kind of queries the copilot is responsible for. See below on [deployment](https://www.notion.so/AI-Copilots-User-Guide-IFRS-1-a66d7c4f581f471299fc7eeb9afbecaa?pvs=21)._

Reviewer: This doesnt seem to link to anything legit, its a random IFRS page?

Reviewer: Okay it looks like there are a few more of this base URL. We shouldn't use this URL but defer to you on what to do about it. Also maybe not even worth blocking on.

![copilot_user_inputs_section.png](/docs/assets/copilots/copilot_user_inputs_section.png)

### f. Tools

Copilots have the ability to use predefined tools. Many of the tools are in Beta, so reach out to [support@credal.ai](mailto:support@credal.ai) if you have any questions or feedback.

![Copilot Tools.png](/docs/assets/copilots/Copilot_Tools.png)

1. **Code Interpreter**
   1. Code Interpreter is primarily used to generate charts and graphs. Your copilot will use the code interpreter tool to write a Python script, and will use that Python script to generate a chart or graph. The chart will be displayed visually in the message response.
   2. This feature is commonly used to plot data that’s extracted from a document or generated by the LLM.
   3. Only available when using OpenAI models.
   4. Streaming responses will be disabled when Code Interpreter is used.

> 2. **Smart Filtering**

Reviewer: A big big TODO here is to talk about AI Extracted Entities, the only self-serve way to use smart filtering, and to contact your credal team about structured data FROM sources

First, click into the channel in slack, and then click on the channel name over the top. In the screenshot below, this is the button called “credal-founders” (this is the name of an example private channel. Your button will have the name of your private channel

![channel_name_in_slack.png](/docs/assets/copilots/channel_name_in_slack.png)

When you do that, you will see the below screen - click the integrations tab as shown below:

![integrations.png](/docs/assets/copilots/integrations.png)

Finally, you should see an option to Add apps to `YOUR CHANNEL NAME`

Search for [Credal.ai](http://Credal.ai) and then click “Add”

![add_apps_to_channel.png](/docs/assets/copilots/add_apps_to_channel.png)

c. API

You can connect your copilot to other applications or build your own tools that route queries to your copilot through a unique API. Use the toggle to enable API access and generate a copilot API key. See our [API documentation](https://docs.credal.ai/api-reference/copilots) for more information on implementation.

![enable_copilot_api_access.png](/docs/assets/copilots/enable_copilot_api_access.png)

> ### c. Custom SlackApp

Reviewer: ```suggestion
### c. Custom Slack Apps (Branding, Name, Profile Picture)
```

> If you want to set up your Copilot as a Custom SlackApp, you can do that. Although the process is a little more involved than using the simple “Deploy Copilot to Slack” button, it does give you some additional benefits, such as the ability for users to DM the bot directly, and also the ability to upload custom images, icons, slash commands etc in Slack. To learn how to do this, see the documentation for [Creating a custom SlackApp in Credal](https://www.notion.so/Credal-Co-Pilots-Slack-App-Manifest-1c36c60bfc9345148e9a087c00cc2e89?pvs=21)

Reviewer: ```suggestion
Credal supports custom Slack Apps, commonly known as Slack "bots", for your individual Copilot.
Although the process is a little more involved than using the simple “Deploy Copilot to Slack” button, it does give you some additional benefits, such as the ability for users to DM the bot directly, and also the ability to upload custom images, icons, slash commands etc in Slack. To learn how to do this, see the documentation for [Creating a custom SlackApp in Credal](https://www.notion.so/Credal-Co-Pilots-Slack-App-Manifest-1c36c60bfc9345148e9a087c00cc2e89?pvs=21)
Contact your Credal team for help with setup. 
```

## 5. **Querying Your Copilot**

You can ask your copilot questions through Credal or via third party applications you have [deployed](https://www.notion.so/AI-Copilots-User-Guide-IFRS-1-a66d7c4f581f471299fc7eeb9afbecaa?pvs=21) it to.

> ### a. **Using copilots through Credal**

Reviewer: ```suggestion
### a. **Hosting Copilots through Credal's UI**
```

To use a copilot in Credal, select your copilot from the dropdown menu in the chat menu.

![chat_with_copilot.png](/docs/assets/copilots/chat_with_copilot.png)

Then simply type your questions in the search bar. Your copilot will return an answer, citing the sources relied upon, which you can click to review.

![copilot_chat_interface_example.png](/docs/assets/copilots/copilot_chat_interface_example.png)

> ### b. Using copilots through t**hird party applications**

Reviewer: ```suggestion
### b. Using Copilots within **third party apps** by API
```
